app/models/idea.rb

```diff
@@ -105,9 +109,34 @@ def approved_by_admin?
     lead
   ]
 
+  def review_year
+    review_date&.strftime('%Y')
+  end
+
+  def review_month
+    review_date&.strftime('%m')
+  end
+
+  def review_day
+    review_date&.strftime('%d')
+  end
+
   private
 
   def send_assigned_user_email
     IdeaMailer.assigned_idea_email_template(assigned_user, self).deliver_now if saved_change_to_assigned_user_id?
   end
+
+  def update_review_date
+    if !@review_year.blank? || !@review_month.blank? || !@review_day.blank?
```

Comment:
This would probably read a bit better if it was `@review_year.present?` instead of checking for blank?